In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statistics as stat

In [3]:
data= pd.read_csv('/home/iswar/Downloads/7. Testing of hypothesis & chi-square test-20220712T140014Z-001/7. Testing of hypothesis _ chi-square test/dataset w7.csv')
data

Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

In [6]:
data.isnull().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

# Carry out descriptive analytics on the dataset.

In [13]:
marks_mean=data['Total Marks'].mean()
marks_stdev=data['Total Marks'].std()
print("Mean mark obtained by the students: ",marks_mean)
print("Standard deviation of mark obtained by the students: ",marks_stdev)

Mean mark obtained by the students:  32.32142857142857
Standard deviation of mark obtained by the students:  12.454520438984368


In [7]:
data['Gender'].value_counts()

female    16
male      12
Name: Gender, dtype: int64

In [14]:
data['Test preparation'].value_counts()

none         21
completed     7
Name: Test preparation, dtype: int64

In [9]:
data.groupby('Gender')['Total Marks'].mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

In [10]:
data.groupby('Test preparation')['Total Marks'].mean()

Test preparation
completed    44.571429
none         28.238095
Name: Total Marks, dtype: float64

In [12]:
data.groupby(['Gender','Test preparation'])['Total Marks'].mean()

Gender  Test preparation
female  completed           45.250000
        none                29.916667
male    completed           43.666667
        none                26.000000
Name: Total Marks, dtype: float64

There are 16 females and 12 males in the class.\
The females scored an average marks of 33.75 while the males scored an average score of 30.424.\
While 7 students had some kind of test preparation, the remaining 21 had none.\
Those with test preparation scored an average score of 44.57 while those without preparation scored an average of 28.23


# Check whether female and male students scored the same marks.

In [15]:
from scipy.stats import ttest_ind

H0 : Males and females scored similar marks.\
H1 : Males and females did not score similar marks.\

In [18]:
data_m = data[data["Gender"] == "male"][["Total Marks"]]
data_f = data[data["Gender"] == "female"][["Total Marks"]]
alpha = 0.05
t_score, p_value = ttest_ind(data_m,data_f)
print('t value : ',t_score)
print('p value : ',p_value)
if p_value > alpha:
    print("Failed to reject null hypothesis")
else:
    print(" Reject null hypothesis")

t value :  [-0.69408853]
p value :  [0.49378223]
Failed to reject null hypothesis


Null hypothesis is accepted in the above case. Males and females scored similar marks.

# Whether test preparation helps the students?

H0: Test Preperation did not help the students.\
H1: Test Preperation helped the students.

In [20]:
testprep_yes=data[data["Test preparation"]=="completed"][["Total Marks"]]
testprep_no=data[data["Test preparation"]=="none"][["Total Marks"]]
alpha=0.05
t_score,p_value=ttest_ind(testprep_yes,testprep_no)
print('t score= ',t_score) 
print('p value= ',p_value)
if(p_value>alpha):
    print("Failed to reject null hypothesis")
                     
else:
    print("Reject null hypothesis")

t score=  [3.61436375]
p value=  [0.00126713]
Reject null hypothesis


Null hypothesis is rejected in the above case. Test preparation helped the students.